meshapp/src/components/SignUp/StepOne.tsx

```diff
@@ -0,0 +1,380 @@
+          </Grid>
+        </Grid>
+      </Grid>
+
+      <Divider sx={{ borderColor: "signUpDivider.main", opacity: 0.5 }} />
+
+      {/*Location Info Section*/}
+      <Grid container justifyContent="center" style={informationSectionStyle}>
+        {/*Title*/}
+        <Grid item xs={9} sm={8} md={6}>
+          <Grid container justifyContent="center">
+            <Grid item xs={12}>
+              <Typography variant="h5" color="text.main">
+                Location Information
+              </Typography>
+            </Grid>
+          </Grid>
+          {/*Countries and States Select*/}
+          <CountriesAndStatesSelect />
+        </Grid>
+      </Grid>
+
+      <Divider sx={{ borderColor: "signUpDivider.main", opacity: 0.5 }} />
+
+      {/*Email, Password, Terms of Service/Email Opt-in Section*/}
+      <Grid container justifyContent="center" style={informationSectionStyle}>
+        <Grid item xs={9} sm={8} md={6}>
+          <Grid container justifyContent="center" columnSpacing={4}>
+            {/*Email*/}
+            <Grid item xs={12} sm={6}>
+              <EmailAddress />
+            </Grid>
+
+            {/*Password*/}
+            <Grid item xs={12} sm={6}>
+              <Password />
+            </Grid>
+
+            {/*Terms and Conditions/Email Opt-in*/}
+            <Grid
+              item
+              xs={12}
+              sm={6}
+              sx={{ color: "text.primary" }}
+              order={{ xs: 3, sm: 2 }}
+            >
+              <FormGroup>
+                <TermsAndConditions />
+
+                <EmailUpdates />
+              </FormGroup>
+            </Grid>
+
+            {/*Confirm Password*/}
+            <Grid item xs={12} sm={6} order={{ xs: 2, sm: 3 }}>
+              <ConfirmPassword />
+            </Grid>
+          </Grid>
+        </Grid>
+      </Grid>
+      <Divider sx={{ borderColor: "signUpDivider.main", opacity: 0.5 }} />
+    </Grid>
+  );
+}
+//---------------------------------------INPUTS-----------------------------------------------------------
+
+//first name text field
+function FirstName() {
+  return (
+    <CustomStandardTextField
+      label="First Name *"
+      id="first-name"
+      fieldName="firstName"
+      required={true}
+    />
+  );
+}
+
+//last name text field
+
+function LastName() {
+  return (
+    <CustomStandardTextField
+      label="Last Name *"
+      id="last-name"
+      required={true}
+      fieldName="lastName"
+    />
+  );
+}
+
+//nickname text field
+
+function Nickname() {
+  return (
+    <CustomStandardTextField
+      label="Nickname"
+      id="nickname"
+      fieldName="nickName"
+    />
+  );
+}
+
+//phone number text field
+
+function PhoneNumber() {
+  return (
+    <CustomStandardTextField
+      label="Phone Number"
+      id="phone-number"
+      fieldName="phoneNumber"
+      validators={{
+        pattern: (value) => {
+          let phoneNumRegex =
+            /^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$/;
```

Comment:
Store this and other string variables as constants somewhere else instead? I don't think we need to allocate these each time, though I'm not sure how TypeScript handles these (it might optimize by making them constants behind the scenes).

Reply:
does it matter whether Im storing these strings at the top of StepOne or if I just put them at the top level of these component functions (like for example putting the phone regex right above the return statement of PhoneNumber())?